src/main/java/io/confluent/connect/jdbc/BulkTableQuerier.java

```diff
@@ -78,6 +78,15 @@ public SourceRecord extractRecord() throws SQLException {
       default:
         throw new ConnectException("Unexpected query mode: " + mode);
     }
+
+    if (keyColumn != null && keySchema != null) {
+      Object key = record.get(keyColumn);
+
+      if (key != null) {
```

Comment:
Why is this structured such that you could fall through to the non-`keyColumn` case? If the key comes back `null` but the `keySchema` isn't optional, shouldn't that cause an error?